### DISCUSSION TITLE
Monin-Obukhov PRofiles - z ref

### DISCUSSION THREAD

**Discussion by fcurow** [firemodels/fds#13554] [General]
To whom it may concern,

Is z_ref taken in absolute or relative terms, eg if the domain is -5m to 5 m in the z direction and there are obstructions from -5 m to 1 m representing the ground and I have a wind value at z = 2m, would the Z_ref in this case be 2m in absolute terms or 1m relative to the obstructions that make up the ground or maybe 7m as it would be relative to the z-min of the domain? The user's guide indicates "If Z_REF is not speciﬁed, it is assumed that the wind speed is taken at a height 2 m off the ground." What is "off the ground" relative to?

Best Regards,
Sal

**Comment by drjfloyd** [comment]
Z_REF is relative to GROUND_LEVEL (default is 0 m)

**Reply by fcurow** [reply]
Sorry, are you guys sure that Z_Ref is relative to GROUND_LEVEL, i.e. should be specified as a delta relative to the ground_level value, and not an absolute z value relative to the global coordinate system? The initial flow values I am seeing suggest it should be an absolute, not a relative value.

**Reply by mcgratta** [reply]
there may be a bug. It looks like the M-O profiles are based on `Z_REF` absolute, whereas elsewhere, `Z_REF` is relative to `GROUND_LEVEL`.

**Reply by fcurow** [reply]
okay, thank you!

**Comment by fcurow** [comment]
Thank you, based on the example provided, does it make sense to adjust the GROUND_LEVEL variable to 1m? Would there be issues otherwise?

**Comment by mcgratta** [comment]
For a domain that is 10 m high, I wonder why you are using M-O theory or profiles. Is this a wind tunnel you are looking at? We typically make `GROUND_LEVEL=0` by default and leave it that way. If you do change it, pay close attention to the resulting flow field because we've never really used M-O for such a small domain, and when there are obstructions and unlevel terrain, the basic premise of M-O may not be appropriate.

**Comment by fcurow** [comment]
My domain is much larger, this is just an example to make it easier to discuss.

**Comment by mcgratta** [comment]
OK, but when we implemented the M-O profiles, we may have made some assumptions about the size of the domain. For example, surface `ROUGHNESS` and similar parameters are based on the assumption that you are modeling hundreds of meters into the atmosphere, and also that ground level details such as obstructions and terrain are accounted for via these various length scales like L. We've seen some users send in cases where the M-O profiles and assumptions are not consistent with the size and scope of the simulation.

**Comment by fcurow** [comment]
Okay, if my domain is 63 m high, what method do you recommend for implementing a velocity profile?

**Comment by mcgratta** [comment]
I like M-O profiles for relatively flat terrain where the external lateral boundaries are sufficiently far from the "action" to allow the flow over terrain or obstructions to develop naturally. How far away is hard to say. Just keep in mind that with M-O you forcing the wind to have a certain temperature and velocity profile at the upstream boundary. These profiles are based on an assumption that the upstream terrain is of a certain roughness and temperature relative to the air stream. 

All this being said, I think it is better to specify a `PRESSURE_GRADIENT_FORCE` in units of Pa/m. This parameter adds a force term to the momentum equation that drives the air in a particular direction much like wind is driven in the atmosphere by gradients of barometric pressure. With this, you should be able to specify boundary conditions at the surface, like a temperature or heat flux and surface roughness, that should produce a velocity and temperature profile that would be similar to the M-O profile. This is a bit like the difference between specifying a heat release rate or predicting it. Predicting it is more satisfying but that comes with a cost. It is hard to just dial in a value for the `PRESSURE_GRADIENT_FORCE` such that you will get back exactly what might have been measured at a near-ground weather station.  But if you are doing research on the nature of the atmospheric boundary layer near the ground and its effect on, say, a building or a wildfire, I think it best to consider the `PGF`. Just specifying an M-O profile may be inconsistent with whatever else you are simulating.

**Reply by fcurow** [reply]
Thank you

**Comment by mcgratta** [comment]
I am creating an [issue](https://github.com/firemodels/fds/issues/13564).
